### PR TITLE
Notifiarr updates & other pulls

### LINKED ISSUE #844: Pull Sonarr commit 'Fixed: Cleanse Tracker Announce Keys from logs'
Pull https://github.com/Sonarr/Sonarr/commit/e67864fecb4a66792152ff8fb0889ee1a176919f

```
git cherry-pick -ex e67864fecb4a66792152ff8fb0889ee1a176919f
```

This commit has 2 conflict(s)

You can ask me to `close all`, `open pr here` or `open pr everywhere`

### LINKED ISSUE #937: Pull Radarr commit 'Fixed: Cleanse Notifiarr APIKey from logs'
Pull https://github.com/Radarr/Radarr/commit/ea3a6171fa76bf571114d8f28c8e0b4cb56c057b

```
git cherry-pick -ex ea3a6171fa76bf571114d8f28c8e0b4cb56c057b
```

This commit has 2 conflict(s)

You can ask me to `close all`, `open pr here` or `open pr everywhere`

### LINKED ISSUE #1014: Pull Radarr commit 'New: DiscordNotifier is now Notifiarr (#6263)'
Pull https://github.com/Radarr/Radarr/commit/27e871656e999c02e777099f8b101c627fdd33f8

```
git cherry-pick -ex 27e871656e999c02e777099f8b101c627fdd33f8
```

This commit has 3 conflict(s)

You can ask me to `close all`, `open pr here` or `open pr everywhere`

### PR DESCRIPTION
**Pull request by bakerboy448**
#### Database Migration
YES - 7

#### Description
A few sentences describing the overall goals of the pull request's commits.

#### Todos
- [X] Tests
- [X] Wiki Updates

#### Issues Fixed or Closed by this PR

* Fixes #1014 
* Fixes #844 
* Fixes #937

### CODE REVIEW

File: src/NzbDrone.Common.Test/InstrumentationTests/CleanseLogMessageFixture.cs [hunk 1]
```diff
@@ -12,7 +12,7 @@ public class CleanseLogMessageFixture
         [TestCase(@"http://rss.torrentleech.org/mySecret")]
         [TestCase(@"http://rss.torrentleech.org/rss/download/12345/01233210/filename.torrent")]
         [TestCase(@"http://www.bitmetv.org/rss.php?uid=mySecret&passkey=mySecret")]
-        [TestCase(@"https://rss.omgwtfnzbs.org/rss-search.php?catid=19,20&user=Readarr&api=mySecret&eng=1")]
+        [TestCase(@"https://rss.omgwtfnzbs.org/rss-search.php?catid=19,20&user=sonarr&api=mySecret&eng=1")]
```
Comment (RetroDadson) [inline src/NzbDrone.Common.Test/InstrumentationTests/CleanseLogMessageFixture.cs]: Sonarr?

File: src/NzbDrone.Common.Test/InstrumentationTests/CleanseLogMessageFixture.cs [hunk 2]
```diff
@@ -72,16 +80,6 @@ public void should_clean_message(string message)
...
-        //GoodReads
-        [TestCase(@"{""signatureMethod"": ""hmacSha1"",""signatureTreatment"": ""escaped"",""type"": ""protectedResource"",""method"": ""GET"",""token"": ""mytoken"",""tokenSecret"": ""mytokensecret"",""requestUrl"": ""https://www.goodreads.com/review/list.xml"",""parameters"": {  ""_nc"": ""1"",  ""v"": ""2"",  ""id"": ""999999999"",  ""shelf"": ""currently-reading"",  ""per_page"": ""200"",  ""page"": ""1""}")]
-        public void should_cleanGoodRead_message(string message)
-        {
-            var cleansedMessage = CleanseLogMessage.Cleanse(message);
-
-            cleansedMessage.Should().NotContain("mytokensecret");
-            cleansedMessage.Should().NotContain("mytoken");
-        }
-
```
Comment (RetroDadson) [inline src/NzbDrone.Common.Test/InstrumentationTests/CleanseLogMessageFixture.cs]: why removing goodreads test?

File: src/NzbDrone.Common/Instrumentation/CleanseLogMessage.cs [hunk 3]
```diff
@@ -42,10 +46,7 @@ public class CleanseLogMessage
                 new Regex(@"(?<=\?|&)(authkey|torrent_pass)=(?<secret>[^&=]+?)(?=""|&|$)", RegexOptions.Compiled | RegexOptions.IgnoreCase),
 
                 // Plex
-                new Regex(@"(?<=\?|&)(X-Plex-Client-Identifier|X-Plex-Token)=(?<secret>[^&=]+?)(?= |&|$)", RegexOptions.Compiled | RegexOptions.IgnoreCase),
-
-                // Good Reads
-                new Regex(@"(?<=""(token|tokensecret)"":\s)""(?<secret>[^""]+?)""", RegexOptions.Compiled | RegexOptions.IgnoreCase)
+                new Regex(@"(?<=\?|&)(X-Plex-Client-Identifier|X-Plex-Token)=(?<secret>[^&=]+?)(?= |&|$)", RegexOptions.Compiled | RegexOptions.IgnoreCase)
```
Comment (RetroDadson) [inline src/NzbDrone.Common/Instrumentation/CleanseLogMessage.cs]: again goodreads removal?

**Review by RetroDadson — CHANGES_REQUESTED** [review]
small bits around removing goodreads tests

File: src/NzbDrone.Common/Instrumentation/CleanseLogMessage.cs [hunk 4]
```diff
@@ -41,11 +45,11 @@ public class CleanseLogMessage
...
 
-                // Plex
-                new Regex(@"(?<=\?|&)(X-Plex-Client-Identifier|X-Plex-Token)=(?<secret>[^&=]+?)(?= |&|$)", RegexOptions.Compiled | RegexOptions.IgnoreCase),
-
                 // Good Reads
-                new Regex(@"(?<=""(token|tokensecret)"":\s)""(?<secret>[^""]+?)""", RegexOptions.Compiled | RegexOptions.IgnoreCase)
+                new Regex(@"(?<=""(token|tokensecret)"":\s)""(?<secret>[^""]+?)""", RegexOptions.Compiled | RegexOptions.IgnoreCase),
+
+                // Plex
+                new Regex(@"(?<=\?|&)(X-Plex-Client-Identifier|X-Plex-Token)=(?<secret>[^&=]+?)(?= |&|$)", RegexOptions.Compiled | RegexOptions.IgnoreCase)
```
Comment (RetroDadson) [inline src/NzbDrone.Common/Instrumentation/CleanseLogMessage.cs]: We don't need Plex token cleansing, no ebook support in Plex.

File: src/NzbDrone.Common.Test/InstrumentationTests/CleanseLogMessageFixture.cs [hunk 5]
```diff
@@ -54,16 +54,24 @@ public class CleanseLogMessageFixture
         [TestCase(@"getTorrents(""mySecret"", [asdfasdf], 100, 0)")]
         [TestCase(@"""DownloadURL"":""https:\/\/broadcasthe.net\/torrents.php?action=download&id=123&authkey=mySecret&torrent_pass=mySecret""")]
 
-        // Spotify Refresh
-        [TestCase(@"https://auth.servarr.com/renew?refresh_token=mySecret")]
-
         // Plex
         [TestCase(@" http://localhost:32400/library/metadata/12345/refresh?X-Plex-Client-Identifier=1234530f-422f-4aac-b6b3-01233210aaaa&X-Plex-Product=Sonarr&X-Plex-Platform=Windows&X-Plex-Platform-Version=7&X-Plex-Device-Name=Sonarr&X-Plex-Version=3.0.3.833&X-Plex-Token=mySecret")]
```
Comment (RetroDadson) [inline src/NzbDrone.Common.Test/InstrumentationTests/CleanseLogMessageFixture.cs]: Don't need Plex test, can you add a goodreads test though please.

**Review by RetroDadson — CHANGES_REQUESTED** [review]
Just a couple more changes, then it should be good to go

**Comment by RetroDadson** [comment]
Oh and a rebase please!

File: src/NzbDrone.Common.Test/InstrumentationTests/CleanseLogMessageFixture.cs (hunk 5 above)
Comment (bakerboy448) [reply src/NzbDrone.Common.Test/InstrumentationTests/CleanseLogMessageFixture.cs]: good reads exists; see further down in the fixture

File: src/NzbDrone.Common/Instrumentation/CleanseLogMessage.cs [hunk 6]
```diff
@@ -41,11 +45,9 @@ public class CleanseLogMessage
                 new Regex(@"getTorrents\(""(?<secret>[^""]+?)""", RegexOptions.Compiled | RegexOptions.IgnoreCase),
                 new Regex(@"(?<=\?|&)(authkey|torrent_pass)=(?<secret>[^&=]+?)(?=""|&|$)", RegexOptions.Compiled | RegexOptions.IgnoreCase),
 
-                // Plex
-                new Regex(@"(?<=\?|&)(X-Plex-Client-Identifier|X-Plex-Token)=(?<secret>[^&=]+?)(?= |&|$)", RegexOptions.Compiled | RegexOptions.IgnoreCase),
-
                 // Good Reads
                 new Regex(@"(?<=""(token|tokensecret)"":\s)""(?<secret>[^""]+?)""", RegexOptions.Compiled | RegexOptions.IgnoreCase)
+
```
Comment (RetroDadson) [inline src/NzbDrone.Common/Instrumentation/CleanseLogMessage.cs]: Remove this blank line

**Review by RetroDadson — CHANGES_REQUESTED** [review]
One last final thing

**Review by ta264 — APPROVED** [review]

**Pull request merged**